app/pages/home-for-user/index.jsx

```diff
@@ -1,75 +1,76 @@
 import React from 'react';
 import { Link } from 'react-router';
-import BlurredImage from './blurred-image';
+import qs from 'qs';
 import Pullout from 'react-pullout';
+import BlurredImage from './blurred-image';
 import getUserRibbonData from '../../lib/get-user-ribbon-data';
 import CircleRibbon from './circle-ribbon';
 import RecentProjectsSection from './recent-projects';
 import RecentCollectionsSection from './recent-collections';
 import RecentMessagesSection from './recent-messages';
 import MyBuildsSection from './my-builds';
 import ProjectStats from './project-stats';
-import qs from 'qs';
 import HomePageSocial from '../home-not-logged-in/social';
 import NewsSection from './news-pullout';
-
+import mediaActions from '../lab/actions/media';
 
 const SECTIONS = {
   projects: RecentProjectsSection,
   collections: RecentCollectionsSection,
   messages: RecentMessagesSection,
-  builds: MyBuildsSection,
+  builds: MyBuildsSection
 };
 
-const HomePageForUser = React.createClass({
-  propTypes: {
-    user: React.PropTypes.object.isRequired,
-    location: React.PropTypes.object,
-  },
-
-  contextTypes: {
-    setAppHeaderVariant: React.PropTypes.func,
-  },
+export default class HomePageForUser extends React.Component {
+  constructor(props) {
+    super(props);
 
-  getDefaultProps() {
-    return {
-      user: {},
-    };
-  },
-
-  getInitialState() {
-    return {
+    this.state = {
+      buttonFocus: false,
       backgroundSrc: '',
       avatarSrc: '',
       showNews: false,
       ribbonData: [],
       loading: false,
       error: null,
+      pendingMedia: [],
       selectedProjectID: null,
-      openSection: null,
+      openSection: null
     };
-  },
+
+    this.toggleNews = this.toggleNews.bind(this);
+    this.toggleFocus = this.toggleFocus.bind(this);
+    this.updateBackground = this.updateBackground.bind(this);
+    this.createLinkedResource = this.props.actions.createLinkedResource.bind(this);
+    this.uploadMedia = this.props.actions.uploadMedia.bind(this);
+  }
 
   componentDidMount() {
+    const backgroundBtn = document.getElementById('change-background');
+    backgroundBtn.addEventListener('focus', this.toggleFocus, true);
+    backgroundBtn.addEventListener('blur', this.toggleFocus, true);
```

Comment:
Is there a reason why these event listeners are being added this way instead of directly on the element? If so, then I'd prefer using a ref to be consistent with using the virtual DOM instead of directly manipulating the DOM.